tools/isobuild/linker.js

```diff
@@ -145,32 +150,121 @@ _.extend(Module.prototype, {
     // Prologue
     var chunks = [];
 
+    const result = {
+      // This object will be populated with .source, .servePath,
+      // .sourceMap, and (optionally) .exportsName properties before being
+      // returned from this method in a singleton array.
+      servePath: self.combinedServePath,
+    };
+
     // Emit each file
-    _.each(self.files, function (file) {
-      if (!_.isEmpty(chunks)) {
-        chunks.push("\n\n\n\n\n\n");
+    if (self.useMeteorInstall) {
+      const tree = {};
+
+      _.each(self.files, function (file) {
+        if (file.bare) {
+          // Bare files will be added in between the synchronous require
+          // calls below.
+          return;
+        }
+
+        const parts = file.installPath.split("/");
```

Comment:
this code is very technical. i'm about to dig in to understand what it does but this should definitely have a comment above it in "human english" explaining what it does.